tower-http/src/cors.rs

```diff
@@ -0,0 +1,627 @@
+//! Middleware which adds headers for [CORS][mdn].
+//!
+//! # Example
+//!
+//! ```
+//! use http::{Request, Response, Method, header};
+//! use hyper::Body;
+//! use tower::{ServiceBuilder, ServiceExt, Service};
+//! use tower_http::cors::{CorsLayer, Any};
+//! use std::convert::Infallible;
+//!
+//! async fn handle(request: Request<Body>) -> Result<Response<Body>, Infallible> {
+//!     Ok(Response::new(Body::empty()))
+//! }
+//!
+//! # #[tokio::main]
+//! # async fn main() -> Result<(), Box<dyn std::error::Error>> {
+//! let cors = CorsLayer::new()
+//!     .allow_methods(vec![Method::GET, Method::POST, Method::OPTIONS])
+//!     .allow_origin(Any)
+//!     .allow_credentials(false);
```

Comment:
it could be nice to have comments explaining what the different configuration methods are doing here?